### PR TITLE
Commoncode

### PR DESCRIPTION
**Pull request by typelogic**
## Issue

Provide common code base repository for Java and Android.

https://github.com/idpass/idpass-lite-java/issues/10

## Changes

Add `android` folder from where `aar` Android library is built.

1. Add Android `aar` build
2. Extracted Dlib models outside from native library when building for Android

Test sample [idpass-lite-java-android-0.1.aar](https://drive.google.com/file/d/1gzW6d3yQI0qfOqLColrKhJ50dBzbfO6v/view?usp=sharing) library. And [test example Android usage snippet](https://gist.github.com/typelogic/7188dc188d4a4cccbf1463dd31338fe7)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -8,4 +8,4 @@ jobs:
       - checkout
       - run:
           name: "build & test"
-          command: ./gradlew build
+          command: ./build.sh jar
```
Comment (jeremi) [inline .circleci/config.yml]: The script just call ./gradlew build so let's keep it as is

File: README.md [hunk 2]
```diff
@@ -7,9 +7,13 @@ This is a Java wrapper of the [libidpasslite](https://github.com/idpass/idpass-l
 ![id front](testdata/idpass-lite-java-sample-front.png?raw=true "front") ![id back](testdata/idpass-lite-java-sample-back.png?raw=true "back")
 
 
-## Building
+## Building for Java `idpass-lite-java-${VERSION}.jar`
 ```bash
-./gradlew build
+./build.sh jar
```
Comment (jeremi) [inline README.md]: The script just call ./gradlew build, so let's keep it as it ease. no need to add abstraction

File: README.md [hunk 3]
```diff
@@ -7,9 +7,13 @@ This is a Java wrapper of the [libidpasslite](https://github.com/idpass/idpass-l
...
+```
+## Building for Android `idpass-lite-java-${VERSION}.aar`
+```bash
+./build.sh aar
```
Comment (jeremi) [inline README.md]: let's just do the cd and `./gradlew build`

File: src/main/java/org/idpass/lite/Card.java [hunk 4]
```diff
@@ -47,21 +38,57 @@
...
     private HashMap<String, Object> cardDetails = new HashMap<String, Object>();
     private HashMap<String, String> cardExtras = new HashMap<String, String>();
 
+    /**
+     * Sets the number of pixels per module to scale up the QR code image. A
+     * value of 3 is visually fine.
+     *
+     * @param scale Number of pixels per module
+     */
+
```
Comment (jeremi) [inline src/main/java/org/idpass/lite/Card.java]: I don't think there should be a new line here. Do you use a linter?

**Review by jeremi — CHANGES_REQUESTED** [review]

File: src/main/java/org/idpass/lite/Card.java [hunk 5]
```diff
@@ -47,21 +38,68 @@
     private boolean isAuthenticated = false;
     private byte[] cardPublicKey = null;
     private byte[] cardAsByte = null;
+    private int scale = 1;  // scale of 1 is best for unit tests
```
Comment (jeremi) [inline src/main/java/org/idpass/lite/Card.java:41]: Why do we optimize for unit test and not for normal usage?

**Review by jeremi — CHANGES_REQUESTED** [review]
- Why Is the folder called `app` and not `aar`? 
- Why do we have layout defined?

File: src/main/java/org/idpass/lite/Card.java (hunk 5 above)
Comment (typelogic) [reply src/main/java/org/idpass/lite/Card.java:41]: - `app` is the default folder name when Android project created
- layout.xml is now removed

A scale of 1 corresponds to 1 pixel per module and runs perfectly without intermittent error during unit tests. Scaling it up will result in very intermittent false alert unit test error (1 out of 50).  I tried to research more and found the explanation [here](https://github.com/journeyapps/zxing-android-embedded/issues/395)

**Comment by typelogic** [comment]
> * Why Is the folder called `app` and not `aar`?
> * Why do we have layout defined?

- app is the default folder name when Android project created
- layouts xml is now removed

A scale of 1 corresponds to 1 pixel per module and runs perfectly without intermittent error during unit tests. Scaling it up will result in very intermittent false alert unit test error (1 out of 50). 
I tried to research more and found the explanation [here](https://github.com/journeyapps/zxing-android-embedded/issues/395)

**Comment by jeremi** [comment]
@rjmangubat23 Is it the best practice to put the `aar` code in an app folder? Can you check how other projects are doing?

File: android/app/src/test/java/org/idpass/lite/ExampleUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package org.idpass.lite;
```
Comment (jeremi) [inline android/app/src/test/java/org/idpass/lite/ExampleUnitTest.java]: Can we remove the sample unit test?

**Review by jeremi — CHANGES_REQUESTED** [review]

**Comment by rjmangubat23** [comment]
> @rjmangubat23 Is it the best practice to put the `aar` code in an app folder? Can you check how other projects are doing?

Hello @jeremi after checking other projects and how we named the SmartScanner repo projects, I agree that we shouldn't put the `aar` code in app folder. @typelogic I would suggest to refactoring the `app` folder to a more specific name like `aar` as @jeremi suggested?

Just to note, in renaming/refactoring in Android: 
- Refactor the directory name
- Update change name in `settings.gradle` 

Please see sample projects: 
https://github.com/journeyapps/zxing-android-embedded
https://github.com/journeyapps/zxing-android-embedded/blob/master/settings.gradle

**Pull request merged**